sql: return a SQL Notice if "CREATE TYPE IF NOT EXISTS" command is used to create a type and the type's already existed.

mneverov opened the pull request:
sql: return a SQL Notice if "CREATE TYPE IF NOT EXISTS" command is used to create a type and the type's already existed.

Previously, a command "CREATE TYPE IF NOT EXISTS" returned "CREATE TYPE" message
even if the type had already existed. This patch changes the message to a notice
"NOTICE: type already exists, skipping" to inform the user about a no-op and to
be compliant with PG CREATE TABLE command.

Fixes #62074 

Release note (sql change): return a SQL Notice if "CREATE TYPE IF NOT EXISTS"
command is used to create a type and the type's already existed.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63326)
<!-- Reviewable:end -->

mneverov commented:
This PR closes #62074.
Sorry for the mess.

arulajmani reviewed (changes requested):
Thanks for taking this on! Looks good modulo one minor comment.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63326#-:-MXnSI5m6sqJRTL4kmya:b8ee0lw)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mneverov)

___
*[pkg/sql/create_type.go, line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63326#-MXnRnBe7tCIGtUM_Rfh:-MXnRnBfBfgkpk_Vfcr9:bibwuuh) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f3db8f210325608662ec5d160cfaadf614627fc/pkg/sql/create_type.go#L93)):*
> ```Go
> 	// CREATE TABLE IF NOT EXISTS by checking the return code
> 	// (pgcode.DuplicateRelation) of getCreateTableParams. However, there isn't
> 	// a pgcode for duplicate types, only the more general pgcode.DuplicateObject.
> ```

Would it make sense to re-word this comment to reflect we're buffering a notice instead of doing nothing, as opposed to completely removing it?


<!-- Sent from Reviewable.io -->

mneverov committed:
sql: return a SQL Notice if "CREATE TYPE IF NOT EXISTS" command is used to
create a type and the type's already existed.

Previously, a command "CREATE TYPE IF NOT EXISTS" returned "CREATE TYPE" message
even if the type had already existed. This patch changes the message to a notice
"NOTICE: type already exists, skipping" to inform the user about a no-op and to
be compliant with PG CREATE TABLE command.

Fixes #62074

Release note (sql change): return a SQL Notice if "CREATE TYPE IF NOT EXISTS"
command is used to create a type and the type's already existed.

mneverov reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63326)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani)

___
*[pkg/sql/create_type.go, line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/63326#-MXnRnBe7tCIGtUM_Rfh:-MXp6qEV-7DnJFXSDv5w:bvtef5p) ([raw file](https://github.com/cockroachdb/cockroach/blob/8f3db8f210325608662ec5d160cfaadf614627fc/pkg/sql/create_type.go#L93)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Would it make sense to re-word this comment to reflect we're buffering a notice instead of doing nothing, as opposed to completely removing it?
</blockquote></details>

You're right. Returned the comment back, and mention the buffering a notice.


<!-- Sent from Reviewable.io -->

arulajmani reviewed (approved):
Thanks, this looks good to go :tada: 

bors r=arulajmani

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/63326#-:-MXrLrg5DKzfmlAhfMvK:b-ra5yb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mneverov)



<!-- Sent from Reviewable.io -->

knz commented:
@arulajmani I think it would be useful if you issued a backport PR for this

arulajmani commented:
Makes sense! 21.1.1 and not 21.1.0 though, right?

knz commented:
yes
